pydicom/dataset.py

```diff
@@ -1746,44 +1827,46 @@ def _pretty_str(self, indent=0, top_level_only=False):
 
         # Display file meta, if configured to do so, and have a non-empty one
         if (
-            hasattr(self, "file_meta")
-            and self.file_meta is not None
-            and len(self.file_meta) > 0
+            hasattr(self, "file_meta") and self.file_meta
```

Comment:
`self.file_meta is not None and len(self.file_meta) > 0` are functionally equivalent to `bool(Dataset())`